Code/Framework/AzCore/Tests/Serialization/Json/MapSerializerTests.cpp

```diff
@@ -497,6 +497,31 @@ namespace JsonSerializationTests
         EXPECT_STRCASEEQ("World", entry->second.c_str());
     }
 
+    TEST_F(JsonMapSerializerTests, Load_DuplicateKey_EntryUpdated)
+    {
+        using namespace AZ::JsonSerializationResult;
+
+        this->m_deserializationSettings->m_updateAssociativeContainer = true;
+        this->ResetJsonContexts();
+
+        m_jsonDocument->Parse(R"(
+            {
+                "Hello": "World",
+                "Hello": "Other"
+            })");
+        ASSERT_FALSE(m_jsonDocument->HasParseError());
+
+        StringMap values;
+        ResultCode result = m_unorderedMapSerializer.Load(&values, azrtti_typeid(&values), *m_jsonDocument, *m_jsonDeserializationContext);
+
+        EXPECT_EQ(Processing::Completed, result.GetProcessing());
+        EXPECT_EQ(Outcomes::Success, result.GetOutcome());
+
+        auto entry = values.find("Hello");
+        ASSERT_NE(values.end(), entry);
+        EXPECT_STRCASEEQ("Other", entry->second.c_str());
```

Comment:
FYI WE have added Googletest [Printers](https://github.com/o3de/o3de/blob/development/Code/Framework/AzTest/AzTest/Printers.h#L14-L39) for our AZStd string types and Path types.

So `EXPECT`_EQ` can be used directly here.

Reply:
I'll update

Reply:
Using `EXPECT_EQ` here compares pointers instead of strings.

Reply:
You have to remove the `._cstr()` otherwise it would be comparing to `const char*`